Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleWindow.cpp

```diff
@@ -94,6 +95,10 @@ namespace ShaderManagementConsole
             shaderFullPath.ReplaceExtension("shadervariantlist");
             result = shaderFullPath;
         }
+        else if (msgBox.clickedButton() == cancelBtn || ret == QMessageBox::Close)
+        {
+            result = "";
+        }
 
         return result.Native();
```

Comment:
isn't it bad that this returns result.Native(). the result goes out of scope?

Reply:
It returns by value, so this will copy the string. It would only be a problem if the return type was string&